lib/system/repr.nim

```diff
@@ -22,7 +22,22 @@ proc reprPointer(x: pointer): string {.compilerproc.} =
 
 proc `$`(x: uint64): string =
   var buf: array [0..59, char]
-  discard c_sprintf(buf, "%llu", x)
+  var i = 0
+
+  var n = x
+  while n != 0:
+    let nn = n div 10.uint64
+    buf[i] = char(n - 10.uint64 * nn + ord('0'))
+    inc i
+    n = nn
+
+  let half = int(i div 2)
+  # Reverse
+  for t in 0 .. < half: swap(buf[t], buf[i-t-1])
+
+  if x == 0:
```

Comment:
If you check for 'x == 0' you might as well move the check to the start and have the rest in an 'else'